lib/api/assertions/attributeMatches.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Check if an elements attribute value matches a regex
+ *
+ * @example
+ * 
+ * ```
+ *    this.demoTest = function (browser) {
+ *      browser.assert.attributeMatches('body', 'data-attr', '(value)');
+ *    };
+ * ```
+ *
+ * @method attributeMatches
+ * @param {string|object} definition The selector (CSS/Xpath) used to locate the element. Can either be a string or an object which specifies [element properties](https://nightwatchjs.org/guide#element-properties).
+ * @param {string} attribute The attribute name
+ * @param {string} regexExpression Regex expression to match attribute value.
+ * @param {string} [msg] Optional log message to display in the output. If missing, one is displayed by default.
+ * @api assertions
+ */
+const {setElementSelectorProps} = require('../../utils');
+
+exports.assertion = function(definition, attribute, regexExpression, msg) {
+  this.options = {
+    elementSelector: true
+  };
+
+  this.expected = function() {
+    return this.negate ? `does not matches '${regexExpression}'` : `matches '${regexExpression}'`;
+  };
+
+  this.formatMessage = function() {
+    const message = msg || `Testing if attribute %s of element %s ${this.negate ? 'doesn\'t matches %s' : 'matches %s'}`;
+
+    return {
+      message,
+      args: [`'${attribute}'`, this.elementSelector, `'${regexExpression}'`]
+    };
+  };
+
+  this.pass = function(value) {
+    const regex = new RegExp(regexExpression);
+
+    return regex.test(value);
+  };
```

Comment:
I think we should use `evaluate` instead of `pass`. check this comment over here: https://github.com/nightwatchjs/nightwatch/blob/6285ebcc8242147d9b47331a368e907d4bf94ba9/lib/api/assertions/_assertionInstance.js#L184

Reply:
Done